code/modules/pool/pool_drain.dm

```diff
@@ -0,0 +1,153 @@
+	desc = "A suction system to remove the contents of the pool, and sometimes small objects. Do not insert fingers."
+	anchored = TRUE
+	/// Active/on?
+	var/active = FALSE
+	/// Filling or draining
+	var/filling = FALSE
+	/// Drain item suction range
+	var/item_suction_range = 2
+	/// Fill mode knock away range
+	var/fill_push_range = 6
+	/// Drain mode suction range
+	var/drain_suck_range = 6
+	/// Parent controller
+	var/obj/machinery/pool/controller/controller
+	/// Cycles left for fill/drain while active
+	var/cycles_left = 0
+	/// Mobs we are swirling around
+	var/list/whirling_mobs
+
+	var/cooldown
+
+/obj/machinery/pool/drain/Initialize()
+	START_PROCESSING(SSfastprocess, src)
+	whirling_mobs = list()
+	return ..()
+
+/obj/machinery/pool/drain/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	controller.linked_drain = null
+	controller = null
+	whirling_mobs = null
+	return ..()
+
+/obj/machinery/pool/drain/proc/is_in_our_pool(atom/A)
+	. = FALSE
+	if(istype(A.loc, /turf/open/pool))
+		var/turf/open/pool/P = A.loc
+		if(P.controller == controller)
+			. = TRUE
+
+// This should probably start using move force sometime in the future but I'm lazy.
+/obj/machinery/pool/drain/process()
+	if(!filling)
+		for(var/obj/item/I in range(min(item_suction_range, 10), src))
+			if(!I.anchored && (I.w_class == WEIGHT_CLASS_SMALL))
+				step_towards(I, src)
+				if((I.w_class == WEIGHT_CLASS_TINY) && (get_dist(I, src) == 0))
+					I.forceMove(controller.linked_filter)
+	if(active)
+		if(filling)
+			if(cycles_left-- > 0)
+				playsound(src, 'sound/effects/fillingwatter.ogg', 100, TRUE)
+				for(var/obj/O in orange(min(fill_push_range, 10), src))
+					if(!O.anchored && is_in_our_pool(O))
+						step_away(O, src)
+				for(var/mob/M in orange(min(fill_push_range, 10), src))		//compiler fastpath apparently?
+					if(!M.anchored && isliving(M) && is_in_our_pool(M))
+						step_away(M, src)
+			else
+				for(var/turf/open/pool/P in controller.linked_turfs)
+					P.filled = TRUE
+					P.update_icon()
+				for(var/obj/effect/waterspout/S in range(1, src))
+					qdel(S)
+				controller.drained = FALSE
+				if(controller.bloody < 1000)
+					controller.bloody /= 2
+				else
+					controller.bloody /= 4
+				controller.update_color()
+				filling = FALSE
+				active = FALSE
+		else
+			if(cycles_left-- > 0)
+				playsound(src, 'sound/effects/pooldrain.ogg', 100, TRUE)
+				playsound(src, "water_wade", 60, TRUE)
+				for(var/obj/O in orange(min(drain_suck_range, 10), src))
+					if(!O.anchored && is_in_our_pool(O))
+						step_towards(O, src)
+				for(var/mob/M in orange(min(drain_suck_range, 10), src))
+					if(isliving(M) && !M.anchored && is_in_our_pool(M))
+						step_towards(M, src)
+						whirl_mob(M)
+						if(ishuman(M) && (get_dist(M, src) <= 1))
+							var/mob/living/carbon/human/H = M
+							playsound(src, pick('sound/misc/crack.ogg','sound/misc/crunch.ogg'), 50, TRUE)
+							if(H.lying)			//down for any reason
+								H.adjustBruteLoss(2.5)
```

Comment:
can this be reduced, this is enough to kill you when you are at full health if you get caught in the drain. I'd rather it put you into soft or maybe hard crit rather than outright killing you if you get stuck for every cycle.